File: 프로그래머스/lv2/42885. 구명보트/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

bool comp(int a, int b) {
    return a > b;
}

int solution(vector<int> people, int limit) {
    sort(people.begin(), people.end(), comp);

    int cnt = 0;
    int left = 0;
    int right = people.size() - 1;

    while (left <= right) {
        if (limit >= people[left] + people[right]) {
            left++;
            right--;
        } else {
            left++;
        }
        cnt++;
    }

    return cnt;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        Arrays.sort(people);
        int front=0; int end=people.length-1;
        while(front<=end){ // 추후 등호 고려
            if(people[front]+people[end] > limit){
                answer++;
                end-=1;
            }else{
                front++;
                end-=1;
                answer++;
            }
        }
        
        return answer;
    }
}